src/shared/components/github-connect/GithubConnect.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useCallback, useState, useEffect } from 'react';
+
+import { Modal, Button } from 'ui';
+
+import { useAuthProvider } from 'core/auth';
+import { useCookiesProvider } from 'core/cookies';
+
+import csx from './GithubConnect.scss';
+
+interface State {
+  success: boolean;
+}
+
+const STATE: State = {
+  success: false
+};
+
+const NAME = 'github-connection';
+const NOT_NOW_VALUE = 'not now';
+
+const GithubConnect = () => {
+  const [state, setState] = useState(STATE);
+  const { success } = state;
+
+  const { user, logInViaGithub } = useAuthProvider();
+
+  const { cookies, setCookies, removeCookie } = useCookiesProvider();
+
+  useEffect(() => {
+    user && user.connectedWithGithub && setState({ success: true });
+  }, [user]);
+
+  const connect = useCallback(() => {
+    setCookies(NAME, user.id);
+    logInViaGithub();
+  }, [user]);
+
+  const clearCookies = useCallback(() => {
+    setState(STATE);
+    removeCookie(NAME);
+  }, []);
+
+  const notNow = useCallback(() => {
+    setCookies(NAME, NOT_NOW_VALUE);
+  }, []);
+
+  const shouldShowScreen =
```

Comment:
visible will be enough for this variable name

Reply:
can be simplified with user && ((!user.connectedWithGithub && cookies[NAME] !== NOT_NOW_VALUE) || cookies[NAME] === user.id && success) -